app/src/components/points-of-interest/IAPP/IAPP-Site.tsx

```diff
@@ -104,89 +124,349 @@ export const IAPPSite: React.FC<IAPPSitePropType> = (props) => {
-                    <TableCell className={classes.cell}>Mechanical ID</TableCell>
-                    <TableCell className={classes.cell}>Common Name</TableCell>
-                    <TableCell className={classes.cell}>Treatment Date</TableCell>
-                    <TableCell className={classes.cell}>Agency</TableCell>
-                    <TableCell className={classes.cell}>Hectares</TableCell>
-                    <TableCell align="center" className={classes.cell}>Mech Method</TableCell>
-                    <TableCell align="center" className={classes.cell}>PaperFile ID</TableCell>
-                    <TableCell className={classes.wideCell}>Comments</TableCell>
-                  </TableRow>
-                </TableHead>
-                <TableBody>
-                  {mechanical_treatments.map((row) => (
-                    <TableRow key={row.MechanicalID}>
-                      <TableCell className={classes.cell}>{ifApplicable(row.MechanicalID)}</TableCell>
-                      <TableCell className={classes.cell}>{ifApplicable(row.CommonName)}</TableCell>
-                      <TableCell className={classes.cell}>{ifApplicable(row.TreatmentDate)}</TableCell>
-                      <TableCell className={classes.cell}>{ifApplicable(row.TreatmentAgency)}</TableCell>
-                      <TableCell className={classes.cell}>{ifApplicable(toDecimalPlaces(row.AreaTreated))}</TableCell>
-                      <TableCell align="center" className={classes.cell}>{ifApplicable(row.MechanicalMethod)}</TableCell>
-                      <TableCell align="center" className={classes.cell}>{ifApplicable(row.PaperFileID)}</TableCell>
-                      <TableCell className={classes.wideCell}>{ifApplicable(row.Comment)}</TableCell>
-                    </TableRow>
-                  ))}
-                </TableBody>
-              </Table>
-            </TableContainer>
-          }
+          <IAPPTable
+            headers={[
+              "Treatment ID",
+              "Mechanical ID",
+              "Common Name",
+              "Treatment Date",
+              "Agency",
+              "Hectares",
+              "Mech Method",
+              "PaperFile ID",
+              {
+                className: classes.wideCell,
+                children: "Comments"
+              }
+            ]}
+            rows={mechanical_treatments ? mechanical_treatments.map((row) => ([
+              row.TreatmentID,
+              row.MechanicalID,
+              row.CommonName,
+              row.TreatmentDate,
+              row.TreatmentAgency,
+              parseFloat(row.AreaTreated).toFixed(4),
+              row.MechanicalMethod,
+              row.PaperFileID,
+              {
+                className: classes.wideCell,
+                children: row.Comment
+              }
+            ])) : []}
+            dropdown={(i) =>
+              mechanical_treatments[i].monitoring === undefined || mechanical_treatments[i].monitoring.length === 0
```

Comment:
can we just do this?
```suggestion
              !mechanical_treatments[i].monitoring || mechanical_treatments[i].monitoring.length === 0
```

Reply:
Should be able to, and then it will catch nulls as well